packages/components/nodes/vectorstores/Postgres_Upsert/Postgres_Upsert.ts

```diff
@@ -0,0 +1,212 @@
+import { ICommonObject, INode, INodeData, INodeOutputsValue, INodeParams } from '../../../src/Interface'
+import { Embeddings } from 'langchain/embeddings/base'
+import { Document } from 'langchain/document'
+import { DataSourceOptions } from 'typeorm'
+import { TypeORMVectorStore, TypeORMVectorStoreDocument } from 'langchain/vectorstores/typeorm'
+import { getBaseClasses, getCredentialData, getCredentialParam } from '../../../src/utils'
+import { flatten } from 'lodash'
+import { Pool } from 'pg'
+
+class PostgresUpsert_VectorStores implements INode {
+    label: string
+    name: string
+    version: number
+    description: string
+    type: string
+    icon: string
+    category: string
+    baseClasses: string[]
+    inputs: INodeParams[]
+    credential: INodeParams
+    outputs: INodeOutputsValue[]
+
+    constructor() {
+        this.label = 'Postgres Upsert Document'
+        this.name = 'postgresUpsert'
+        this.version = 1.0
+        this.type = 'Postgres'
+        this.icon = 'postgres.svg'
+        this.category = 'Vector Stores'
+        this.description = 'Upsert documents to Postgres using pgvector'
+        this.baseClasses = [this.type, 'VectorStoreRetriever', 'BaseRetriever']
+        this.credential = {
+            label: 'Connect Credential',
+            name: 'credential',
+            type: 'credential',
+            description: 'Needed when using Postgres cloud hosted',
```

Comment:
is the credential optional here?

Reply:
Sure not! I don't know why the upsert was different from the existing one